helm.el

```diff
@@ -3667,31 +3667,32 @@ without recomputing them, it should be a list of lists."
            ;; updating.
            (helm-log "Matches: %S"
                      (setq matches (or candidates (helm--collect-matches sources))))
-           ;; If computing matches finished and is not interrupted
-           ;; erase the helm-buffer and render results (Fix #1157).
-           (erase-buffer)             ; [1]
-           (if (cl-loop for source in matches
-                       thereis source)
```

Comment:
This loop is redundant with `(when matches`, it does the same, also it use the name source which is confusing because there is no source in matches.